### ISSUE TITLE
Alerts no longer sending after upgrading from 1.14.8 to 1.15.5

### ISSUE THREAD

**Issue by jeanluclariviere** [coreos/kube-prometheus#291]
**What did you do?**
After upgraded from 1.14.8 to 1.15.5, alerts are no longer sending. The alerts are being triggered, I can see them in the alertmanager dashboard, but nothing is being sent. The same manifests applied to a cluster running 1.14.8 work just fine. Anyway suggestions on what steps I can take next to continue troubleshooting?

**Did you expect to see some different?**
Alerts should continue sending.

* Prometheus Operator version:

0.30

* Kubernetes version information:

1.15.8

* Kubernetes cluster kind:

kubeadm

* Manifests:
alertmanager.yaml (inside of the alertmanager)
```
global:
  resolve_timeout: 5m
route:
  receiver: 'email_alert'
  group_by: ['job']
  group_wait: 30s
  group_interval: 5m
  repeat_interval: 12h
  routes:
  - match:
      alertname: Watchdog
    receiver: 'null'
receivers:
- name: 'null'
- name: 'email_alert'
  email_configs:
  - send_resolved: true
    to: me@example.com
    smarthost: smtp.example.com:25
```

* Additional info:
I've specified these versions in my jsonnet as they were listed as kubernetes 1.15.x compatible.

```
    versions+:: {
      nodeExporter: "v0.18.1",
      kubeStateMetrics: "v1.7.2",
      }
```

edit: i realize i was incorrect, and updated my kube-state-metrics version from 1.6.0 to 1.7.2 to ensure compatibility 1.15.x as per their project https://github.com/kubernetes/kube-state-metrics. I'll monitor to see if this helps with the problem, though i don't think these two changes are related as the alerts are already firing.

**Comment by jeanluclariviere** [comment]
> Are you saying the alerts are not triggering in Prometheus or Alertmanager is not sending them? As in you can see targets scraped correctly and rules being loaded in the Prometheus UI?

Sorry if i wasn't clear - alerts are still being triggered, metrics are still being scrapped correctly and rules are being loaded in the prometheus IU. What seems to have stopped working is the actual sending of the notifications via email.

**Comment by brancz** [comment]
Can you see the alerts in the Alertmanager UI when they fire? Can you share the Alertmanager's logs?

**Comment by jeanluclariviere** [comment]
I just redeployed kube-prometheus and re-loaded the alertmanager config. I can see the alerts being triggered and i can see them in the alertmanager UI. Here are the alertmanager container logs.

```
λ kubectl logs -n monitoring -c alertmanager alertmanager-main-0
level=info ts=2019-10-31T17:05:42.569Z caller=main.go:217 msg="Starting Alertmanager" version="(version=0.19.0, branch=HEAD, revision=7aa5d19fea3f58e3d27dbdeb0f2883037168914a)"
level=info ts=2019-10-31T17:05:42.569Z caller=main.go:218 build_context="(go=go1.12.8, user=root@587d0268f963, date=20190903-15:01:40)"
level=warn ts=2019-10-31T17:05:42.619Z caller=cluster.go:228 component=cluster msg="failed to join cluster" err="3 errors occurred:\n\t* Failed to resolve alertmanager-main-0.alertmanager-operated.monitoring.svc:6783: lookup alertmanager-main-0.alertmanager-operated.monitoring.svc on 10.96.0.10:53: no such host\n\t* Failed to resolve alertmanager-main-1.alertmanager-operated.monitoring.svc:6783: lookup alertmanager-main-1.alertmanager-operated.monitoring.svc on 10.96.0.10:53: no such host\n\t* Failed to resolve alertmanager-main-2.alertmanager-operated.monitoring.svc:6783: lookup alertmanager-main-2.alertmanager-operated.monitoring.svc on 10.96.0.10:53: no such host\n\n"
level=info ts=2019-10-31T17:05:42.619Z caller=cluster.go:230 component=cluster msg="will retry joining cluster every 10s"
level=warn ts=2019-10-31T17:05:42.619Z caller=main.go:308 msg="unable to join gossip mesh" err="3 errors occurred:\n\t* Failed to resolve alertmanager-main-0.alertmanager-operated.monitoring.svc:6783: lookup alertmanager-main-0.alertmanager-operated.monitoring.svc on 10.96.0.10:53: no such host\n\t* Failed to resolve alertmanager-main-1.alertmanager-operated.monitoring.svc:6783: lookup alertmanager-main-1.alertmanager-operated.monitoring.svc on 10.96.0.10:53: no such host\n\t* Failed to resolve alertmanager-main-2.alertmanager-operated.monitoring.svc:6783: lookup alertmanager-main-2.alertmanager-operated.monitoring.svc on 10.96.0.10:53: no such host\n\n"
level=info ts=2019-10-31T17:05:42.619Z caller=cluster.go:623 component=cluster msg="Waiting for gossip to settle..." interval=2s
level=info ts=2019-10-31T17:05:42.666Z caller=coordinator.go:119 component=configuration msg="Loading configuration file" file=/etc/alertmanager/config/alertmanager.yaml
level=info ts=2019-10-31T17:05:42.667Z caller=coordinator.go:131 component=configuration msg="Completed loading of configuration file" file=/etc/alertmanager/config/alertmanager.yaml
level=info ts=2019-10-31T17:05:42.673Z caller=main.go:466 msg=Listening address=:9093
level=info ts=2019-10-31T17:05:44.619Z caller=cluster.go:648 component=cluster msg="gossip not settled" polls=0 before=0 now=1 elapsed=2.00014066s
level=info ts=2019-10-31T17:05:50.620Z caller=cluster.go:648 component=cluster msg="gossip not settled" polls=3 before=1 now=2 elapsed=8.000990102s
level=warn ts=2019-10-31T17:05:57.648Z caller=cluster.go:438 component=cluster msg=refresh result=failure addr=alertmanager-main-1.alertmanager-operated.monitoring.svc:6783
level=warn ts=2019-10-31T17:05:57.654Z caller=cluster.go:438 component=cluster msg=refresh result=failure addr=alertmanager-main-2.alertmanager-operated.monitoring.svc:6783
level=info ts=2019-10-31T17:05:58.621Z caller=cluster.go:640 component=cluster msg="gossip settled; proceeding" elapsed=16.002050443s
level=info ts=2019-10-31T17:06:46.234Z caller=coordinator.go:119 component=configuration msg="Loading configuration file" file=/etc/alertmanager/config/alertmanager.yaml
level=info ts=2019-10-31T17:06:46.235Z caller=coordinator.go:131 component=configuration msg="Completed loading of configuration file" file=/etc/alertmanager/config/alertmanager.yaml
```

**Comment by brancz** [comment]
Just double checking, you are seeing the config you configured alertmanager with in the alertmanager UI as well?

**Comment by jeanluclariviere** [comment]
> Just double checking, you are seeing the config you configured alertmanager with in the alertmanager UI as well?

Yes, that is correct.

**Comment by brancz** [comment]
cc @simonpasquier

**Comment by simonpasquier** [comment]
Can you check the `alertmanager_notifications_total` and `alertmanager_notifications_failed_total` metrics?
Are peers listed in the Status page of Alertmanager UI?

**Comment by jeanluclariviere** [comment]
I checked those metrics and i see 0 failed notifications - which leads me to think it might be an issue at the smtp relay. I'll reach out to our sysadmins to see if they can't see anything in the logs, but I find this surprising, because alerts are still working for my clusters 1.14.8.

As for the peers listed on the status page, I can see 3 endpoints, is that what you're referring to?

```
http://10.244.1.10:9093/api/v1/alerts
http://10.244.2.245:9093/api/v1/alerts
http://10.244.2.246:9093/api/v1/alerts
```

**Comment by simonpasquier** [comment]
> I checked those metrics and i see 0 failed notifications

If the SMTP relay rejects the mails, it should be logged by Alertmanager. Is `alertmanager_notifications_total` greater than 0?

Regarding the cluster status, I was talking about the Alertmanager UI.

Finally there would be no log about an alert being fired in Prometheus. If you run Alertmanager with `--log.level=debug`, you'd see alerts being received from Prometheus.

**Comment by jeanluclariviere** [comment]
Guys, thanks for the help here, it put me on the right track. Fortunately, it was not an issue with kube-prometheus, alert manager or the Prometheus operator. It was a coincidence that changes were made to a distribution list we were using that caused this to break. I should have realized this as soon as I saw the failed notifications were all set still at 0. Closing this issue as the upgrade had nothing to do with this breaking.

Sorry for the trouble!
